core/src/main/java/org/aerogear/mobile/core/http/pinning/CertificatePinningCheck.java

```diff
@@ -67,26 +68,26 @@ public void detachListener() {
 
     /**
      * Perform a check and invoke a listener on response.
-     * 
+     *
      * @param url to which a request will be made
      */
     public void execute(@NonNull final String url) {
         HttpRequest request = this.httpModule.newRequest();
-        request.get(url);
-        HttpResponse httpResponse = request.execute();
+        request.get(url).respondWith(new Responder<HttpResponse>() {
+            @Override
+            public void onResult(HttpResponse httpResponse) {
+                CertificatePinningCheck.this.isSuccess = true;
+                if (CertificatePinningCheck.this.listener != null) {
+                    CertificatePinningCheck.this.listener.onSuccess();
+                }
+                CertificatePinningCheck.this.isComplete = true;
+            }
+
+            @Override
+            public void onException(Exception exception) {
+                CertificatePinningCheck.this.listener.onFailure();
```

Comment:
I think we should check if the listener is null.
What about initialising the listener to a NOOP listener, so that no null check needs to be done?

Reply:
I fixed this since it doesn't change anything outside of the class and could have crashed